services/app-web/src/pages/SubmissionRevisionSummary/SubmissionRevisionSummaryV2.tsx

```diff
@@ -49,18 +49,25 @@ export const SubmissionRevisionSummaryV2 = (): React.ReactElement => {
         fetchPolicy: 'network-only',
     })
     const contract = fetchContractData?.fetchContract.contract
-    //We offset version by +1 of index, remove offset to find revision in revisions
+
+    // Offset version by +1 of index, remove offset to find target previous submission in the history list
     const revisionIndex = Number(revisionVersion) - 1
-    const name =
-        contract &&
-        contract?.packageSubmissions.length > Number(revisionVersion)
-            ? contract.packageSubmissions.reverse()[revisionIndex]
```

Comment:
This right here was also part of the problem - we were not using the same logic for the `name` and the data calculations - we should be filtering on CONTRACT_SUBMISSION also here. I changed things to calculate the target previous submission once and use that for both the name and the data calculations for the page